runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -1204,7 +1205,14 @@ TR::CompilationInfoPerThreadRemote::getCachedResolvedMethod(TR_ResolvedMethodKey
       if (*resolvedMethod)
          {
          if (unresolvedInCP)
-            *unresolvedInCP = false;
+            {
+            *unresolvedInCache = methodCacheEntry.unresolvedInCP;
+            if (methodCacheEntry.unresolvedInCP == TR_maybe)
+               {
+               return false;
+               }
+            *unresolvedInCP = (methodCacheEntry.unresolvedInCP == TR_yes) ? true : false;
```

Comment:
Also, setting the parameter `unresolvedInCache` is not needed in this method, since it's only useful when cache verification is enabled, otherwise just having `unresolvedInCP` set is sufficient. 
Instead, you should move `unresolvedInCache` to `getCachedMethodInfo`, replacing `unresolvedInCP` there.